src/adapter/mod.rs

```diff
@@ -37,3 +36,37 @@ pub trait Adapter: Send + Sync {
         field_values: Vec<String>,
     ) -> Result<bool>;
 }
+
+#[cfg(feature = "filtered-adapter")]
+#[derive(Clone, Debug)]
+pub struct Filter {
+    pub p: Vec<String>,
+    pub g: Vec<String>,
+}
+
+#[cfg(feature = "filtered-adapter")]
+#[async_trait]
+pub trait FilteredAdapter: Adapter {
+    async fn load_filtered_policy(&mut self, m: &mut dyn Model, _f: Option<Filter>) -> Result<()>;
+
+    fn is_filtered(&self) -> bool;
+}
+
+#[cfg(feature = "filtered-adapter")]
+#[async_trait]
+impl<T> FilteredAdapter for T
+where
+    T: Adapter,
+{
+    default async fn load_filtered_policy(
```

Comment:
This feature only exists in nightly. It's only used by `FileAdapter` to do some test, we can just remove these lines.

Reply:
If we remove these lines, `DieselAdapter` wont be able to implement `FilteredAdapter` trait

Reply:
Why? you can add 
```rust
impl FilteredAdapter for DieselAdapter {}
```

Reply:
@xcaptain Ok I misunderstood.. I thought you would like to remove `default` which is specialization.

Indeed we can also remove these lines, but in that case we need to merge these two traits.

Reply:
Basically it's not possible to have two traits and switch between them...we always need to pick:

https://github.com/casbin/casbin-rs/pull/119#discussion_r410341021

Reply:
I saw your changes, remove `filtered-adapter` makes it build on stable, but if people needs to use `filtered-adapter` they have to use nightly, can we make `filtered-adapter` work on stable? It's not worthy to switch the whole system's build environment to nightly just because of a small feature in casbin